### PR TITLE
Integrate API for Metrics Conditions

### PR DESCRIPTION
**Pull request by victorhuangwq**
This integrates the update_current_metrics and get_current_metrics endpoint with the condition metrics tab of the metrics feature

## Description

image

1. Initializes the metrics form based on the current settings of the user
1. Calls update_current_metrics with appropriate json upon saving
1. Gives appropriate feedback and form validation

## Motivation and Context

This is part of the metrics feature being developed

## How Has This Been Tested?

It has been tested with the following steps, starting off with a blank setting:
1. Checked grace days option, with 2 grace days by November 30. Save, then refresh page and see that the selection has persisted
1. Checked grades drop option, with 101 percent, 2 consecutive. Gets appropriate error message stating that 101 percent is out of range. Change 101 to 30, then save and refresh page to see that the selection has persisted
1. Checked no submit option, with 3 assignments, and low grades option with 2 assignments and -1 percentage. Gets appropriate error message stating that -1 is out of range. Change -1 to 50, then save and refresh page to see that selection has persisted.
1. Unselects all checkbox, save, and see that selection (or the lack thereof) has persisted.

In all of the above, a successful submit will yield a success banner

Error Situation:
1. Change the update_metrics_endpoint to `update`
1. Save a changed form
1. An error banner should be shown
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [ ] I have updated the documentation accordingly, included in this PR

### CODE REVIEW

File: app/assets/javascripts/metrics.js [hunk 1]
```diff
@@ -1,56 +1,222 @@
...
+	// important to give each element an unique id
+	const message_html = `
+	<div class="ui ${params.type} message" id="message_${message_id}">
+		<i class="close icon"></i>
+		<div class="header">
+		${params.header}
+		</div>
+		${params.message}
+	</div>
+	`;
```
Comment (victorhuangwq) [inline app/assets/javascripts/metrics.js:203]: the main useful thing to utilize in creating HTML elements is template strings (https://hacks.mozilla.org/2015/05/es6-in-depth-template-strings-2/)

It's not supported by I.E at all, but this is just so clean. In fact you can create another function, that generates another HTML string using the template string, and then insert it here. (In this case, imagine params.message is <div><p>hello</p></div> generated using another function)

**Comment by ashleyzhang** [comment]
Everything looks good except one weird error that's happening with the success banners reappearing if you save twice in a row and then try closing the banners.

Steps to replicate: 
1. Fill in fields of two conditions (without checking the boxes yet)
2. Click checkbox for first condition then save
3. Before first success banner closes, click checkbox for second condition then save
4. Try closing both success banners

image

**Review by ashleyzhang — CHANGES_REQUESTED** [review]
Everything looks good except one weird error that's happening with the success banners reappearing if you save twice in a row and then try closing the banners.

Steps to replicate: 
1. Fill in fields of two conditions (without checking the boxes yet)
2. Click checkbox for first condition then save
3. Before first success banner closes, click checkbox for second condition then save
4. Try closing both success banners

image

**Comment by victorhuangwq** [comment]
I have addressed the banner issue accordingly in my latest commit!

**Review by ashleyzhang — APPROVED** [review]
LGTM!

File: app/assets/javascripts/metrics.js [hunk 2]
```diff
@@ -1,56 +1,226 @@
...
+
+});
+
+
+$.getJSON(metrics_endpoints['get'],function(data, status){
+	
+	if(status=='success'){
+
+		// situation when no conditions have been selected
+		if(data.length == 1 && data[0]['condition_type'] == "no_condition_selected")
```
Comment (amartinez116) [inline app/assets/javascripts/metrics.js:64]: Question about this line: when I have no conditions selected I am receiving data.length == 0. Is this supposed to be when instructors have not set up any conditions for their class?

**Pull request merged**